homework-2/src/js/listeners-inputs.js

```diff
@@ -0,0 +1,51 @@
+const checkValidInput = (item) => {
+  item.getElementsByTagName('input')[0].addEventListener('change', (e) => {
+    const value = e.currentTarget.value;
+    const regul = new RegExp(
+      e.currentTarget.getAttribute('type') == 'email'
+        ? /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+        : /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/,
+    );
+
+    if (!regul.test(value)) {
+      item.querySelector('.form-main__input').classList.add('is-error-input');
+      item
+        .getElementsByTagName('label')[0]
+        .classList.replace('is-hidden-error', 'is-visible-error');
+    } else {
+      item.querySelector('.form-main__input').classList.remove('is-error-input');
+      item
+        .getElementsByTagName('label')[0]
+        .classList.replace('animate__backInUp', 'animate__backOutDown');
+
+      setTimeout(() => {
```

Comment:
What for is setTimeout?

Reply:
Post above

Reply:
This setTimeout is nedeed to smothly hide the label